### PR TITLE
sdk: Allow to get stripped state events from the store

### PR DESCRIPTION
**Pull request by zecakeh**
Previously it was not possible to get state events for invited rooms.

This is a pre-requisite for #1979, to be able to use `get_state_events` in `get_members`.

Should we deprecate `StateStore::get_member_event`? It doesn't do anything more than `StateStoreExt::get_state_event_static_for_key` now.

- [x] Public API changes documented in changelogs (optional)

Signed-off-by: Kévin Commaille <zecakeh@tedomum.fr>

### CODE REVIEW

File: crates/matrix-sdk-base/src/deserialized_responses.rs [hunk 1]
```diff
@@ -61,81 +63,200 @@ pub struct MembersResponse {
...
+pub type MemberEvent = SyncOrStrippedState<RoomMemberEventContent>;
+
+impl MemberEvent {
+    /// The inner Content of the wrapped Event.
+    pub fn original_content(&self) -> Option<&RoomMemberEventContent> {
+        match self {
+            MemberEvent::Sync(e) => e.as_original().map(|e| &e.content),
+            MemberEvent::Stripped(e) => Some(&e.content),
+        }
+    }
```
Comment (jplatte) [inline crates/matrix-sdk-base/src/deserialized_responses.rs:251]: Couldn't this be a function on `SyncOrStrippedState<C>` for any `C` (with appropriate bounds)?

File: crates/matrix-sdk-base/src/store/memory_store.rs [hunk 2]
```diff
@@ -376,25 +377,48 @@ impl MemoryStore {
...
         state_key: &str,
-    ) -> Result<Option<Raw<AnySyncStateEvent>>> {
-        Ok(self
+    ) -> Result<Option<RawAnySyncOrStrippedState>> {
+        if let Some(e) = self
+            .stripped_room_state
+            .get(room_id)
+            .as_ref()
+            .and_then(|events| events.get(&event_type))
+            .and_then(|m| m.get(state_key).map(|m| m.clone()))
```
Comment (jplatte) [inline crates/matrix-sdk-base/src/store/memory_store.rs:386]: Is `.cloned()` not applicable here because of an extra deref coercion?

Comment (zecakeh) [reply crates/matrix-sdk-base/src/store/memory_store.rs:386]: Yep, `m` is a `Ref<_, event>`.

**Comment by zecakeh** [comment]
I managed to make the compiler panic :grimacing: .

**Pull request merged**